File: JavaScript/Strings/LeftMostRepeat.js
const leftRepeat = (s) => {
  const count = {};
  for (let i = 0; i < s.length; i++) {
    count[s.charAt(i)] = (count[s.charAt(i)] || 0) + 1;
  }
//   for (let i in count) {
//     console.log(count[i]);
//   }
  for (let i = 0; i < s.length; i++) {
    if (count[s.charAt(i)] === 1) {
      return s.charAt(i);
    }
  }
  return '$';
};

console.log(leftRepeat("abbccddaa"));

File: GFG/Strings/LeftMostRepeat.java
package GFG.Strings;

import java.util.ArrayList;

public class LeftMostRepeat {
    static void leftRepeat(String s) {
        int[] freq = new int[26];
        for (int i = 0; i < s.length(); i++) {
            freq[s.charAt(i) - 'a']++;
        }
        ArrayList<Integer> lis = new ArrayList<>();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            int freqq = --freq[c - 'a'];
            lis.add(freqq);
        }
        for(int i : lis){
            System.out.print(i+" ");
        }

    }

    public static void main(String[] args) {
        String s = "abcacabc";
        leftRepeat(s);
    }
}
